src/indicators/ElementOpticalDepthIndicator.C

```diff
@@ -0,0 +1,107 @@
+/********************************************************************/
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2021 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#ifdef ENABLE_OPENMC_COUPLING
+
+#include "ElementOpticalDepthIndicator.h"
+
+#include "CardinalEnums.h"
+#include "TallyBase.h"
+
+registerMooseObject("CardinalApp", ElementOpticalDepthIndicator);
+
+InputParameters
+ElementOpticalDepthIndicator::validParams()
+{
+  auto params = OpenMCIndicator::validParams();
+  params.addClassDescription(
+      "A class which returns the estimate of a given element's optical depth under the assumption "
+      "that "
+      "elements with a large optical depth experience large solution gradients.");
+  params.addRequiredParam<MooseEnum>("rxn_rate",
+                                     getSingleTallyScoreEnum(),
+                                     "The reaction rate to use for computing the optical depth.");
+  params.addParam<MooseEnum>(
+      "h_type",
+      MooseEnum("min max cube_root", "max"),
+      "The estimate for the length of the element used to compute the optical depth. Options are "
+      "the "
+      "minimum vertex separation (min), the maximum vertex separation (max), and the cube root of "
+      "the element volume (cube_root).");
+
+  return params;
+}
+
+ElementOpticalDepthIndicator::ElementOpticalDepthIndicator(const InputParameters & parameters)
+  : OpenMCIndicator(parameters), _h_type(getParam<MooseEnum>("h_type").getEnum<HType>())
+{
+  std::string score = getParam<MooseEnum>("rxn_rate");
+  std::replace(score.begin(), score.end(), '_', '-');
+
+  // Error check to make sure the score is a reaction rate score and to make sure one of the
+  // [Tallies] has added the score and a flux score.
+  if (!_openmc_problem->isReactionRateScore(score))
+    mooseError(
```

Comment:
Let's try and use `paramError` as much as possible. We haven't had good practices using it consistently, but I'd say let's start now. Can you please change `mooseError`s to `paramError`, if it makes sense?